File: app/src/main/java/bf/fasobizness/bafatech/activities/annonce/ActivityDetailAnnonceUser.java
package bf.fasobizness.bafatech.activities.annonce;

import android.app.ProgressDialog;
import android.content.Intent;
import android.os.Bundle;
import android.util.Log;
import android.view.View;
import android.widget.Button;
import android.widget.LinearLayout;
import android.widget.TextView;
import android.widget.Toast;

import androidx.annotation.NonNull;
import androidx.appcompat.app.AlertDialog;
import androidx.appcompat.app.AppCompatActivity;
import androidx.appcompat.widget.Toolbar;
import androidx.viewpager.widget.ViewPager;

import java.util.ArrayList;
import java.util.List;
import java.util.Objects;

import bf.fasobizness.bafatech.R;
import bf.fasobizness.bafatech.activities.ActivityFullScreen;
import bf.fasobizness.bafatech.adapters.ImagesAdapter;
import bf.fasobizness.bafatech.helper.RetrofitClient;
import bf.fasobizness.bafatech.interfaces.API;
import bf.fasobizness.bafatech.interfaces.OnImageListener;
import bf.fasobizness.bafatech.models.Announce;
import bf.fasobizness.bafatech.models.MyResponse;
import bf.fasobizness.bafatech.utils.MySharedManager;
import retrofit2.Call;
import retrofit2.Callback;
import retrofit2.Response;

public class ActivityDetailAnnonceUser extends AppCompatActivity implements OnImageListener {

    private static final String TAG = "ActivityDetailsAnnonce";
    private ArrayList<String> imagesList;
    private ImagesAdapter imagesAdapter;

    private String user;
    private TextView txt_titre_annonce;
    private TextView txt_text;
    private TextView txt_prix;
    private TextView txt_email;
    private TextView txt_tel1;
    private TextView txt_tel2;
    private TextView txt_tel;
    private TextView txt_location;
    private TextView txt_categorie;
    private TextView txt_updatedAt;
    private TextView txt_date_pub;
    private LinearLayout ann, layout_no_annonce, loading_indicator_ann, layout_ent_offline, layout_busy_system;
    private ViewPager pager;
    private String id_ann;

    private Button btn_edit, btn_delete, btn_edit_photo, btn_share;

    private API api;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_detail_annonce_utilisateur);

        api = RetrofitClient.getClient().create(API.class);
        MySharedManager mySharedManager = new MySharedManager(this);
        user = mySharedManager.getUser();

        Toolbar toolbar = findViewById(R.id.toolbar);
        toolbar.setTitle("");
        setSupportActionBar(toolbar);
        Objects.requireNonNull(getSupportActionBar()).setDisplayHomeAsUpEnabled(true);
        toolbar.setNavigationIcon(R.drawable.left_white);
        toolbar.setNavigationOnClickListener(view -> finish());

        txt_titre_annonce = findViewById(R.id.txt_titre_annonce);
        txt_text = findViewById(R.id.txt_texte_ann);
        txt_prix = findViewById(R.id.txt_prix_annonce);
        txt_email = findViewById(R.id.txt_email_util);
        txt_tel1 = findViewById(R.id.txt_tel1_annonce);
        txt_tel2 = findViewById(R.id.txt_tel2_annonce);
        txt_tel = findViewById(R.id.txt_tel_annonce);
        txt_location = findViewById(R.id.txt_location_annonce);
        txt_categorie = findViewById(R.id.txt_categorie_annonce);
        btn_share = findViewById(R.id.btn_share);
        txt_updatedAt = findViewById(R.id.txt_date_modification);
        txt_date_pub = findViewById(R.id.txt_date_pub);
        pager = findViewById(R.id.flipper_affiche_annonce);

        loading_indicator_ann = findViewById(R.id.loading_indicator_ann);
        layout_ent_offline = findViewById(R.id.layout_ent_offline);
        layout_no_annonce = findViewById(R.id.layout_no_annonce);
        ann = findViewById(R.id.ann);
        layout_busy_system = findViewById(R.id.layout_busy_system);
        Button btn_refresh = findViewById(R.id.btn_refresh);

        layout_ent_offline.setVisibility(View.GONE);

        ann.setVisibility(View.GONE);
        loading_indicator_ann.setVisibility(View.VISIBLE);
        layout_no_annonce.setVisibility(View.GONE);
        layout_busy_system.setVisibility(View.GONE);

        imagesList = new ArrayList<>();
        imagesAdapter = new ImagesAdapter(this, imagesList);
        pager.setAdapter(imagesAdapter);
        imagesAdapter.setOnImageListener(this);

        Intent extras = getIntent();
        if (extras.getStringExtra("id_ann") != null) {
            id_ann = extras.getStringExtra("id_ann");
            btn_refresh.setOnClickListener(v -> getAnnounce(extras.getStringExtra("id_ann")));
        }

        btn_edit = findViewById(R.id.btn_editer);
        btn_delete = findViewById(R.id.btn_delete);
        btn_edit_photo = findViewById(R.id.btn_editer_photo);
        btn_share = findViewById(R.id.btn_share);

    }

    private void getAnnounce(String id_ann) {

        layout_ent_offline.setVisibility(View.GONE);
        loading_indicator_ann.setVisibility(View.VISIBLE);

        Call<Announce.Annonce> call = api.getAnnounce(id_ann, user);
        call.enqueue(new Callback<Announce.Annonce>() {
            @Override
            public void onResponse(@NonNull Call<Announce.Annonce> call, @NonNull Response<Announce.Annonce> response) {
                loading_indicator_ann.setVisibility(View.GONE);
                if (response.isSuccessful()) {
                    Log.d(TAG, response.toString());
                    loading_indicator_ann.setVisibility(View.GONE);
                    ann.setVisibility(View.VISIBLE);
                    Announce.Annonce announce = response.body();
                    if (announce == null) {
                        layout_ent_offline.setVisibility(View.GONE);
                        ann.setVisibility(View.GONE);
                        layout_no_annonce.setVisibility(View.VISIBLE);
                    } else {
                        populateData(announce);
                    }
                } else {
                    layout_busy_system.setVisibility(View.VISIBLE);
                }
            }

            @Override
            public void onFailure(@NonNull Call<Announce.Annonce> call, @NonNull Throwable t) {

                layout_ent_offline.setVisibility(View.VISIBLE);
                ann.setVisibility(View.GONE);
                layout_no_annonce.setVisibility(View.GONE);
                loading_indicator_ann.setVisibility(View.GONE);

                Log.d(TAG, t.toString());
                Toast.makeText(ActivityDetailAnnonceUser.this, getString(R.string.pas_d_acces_internet), Toast.LENGTH_SHORT).show();

            }
        });
    }

    private void populateData(final Announce.Annonce annonce) {
        txt_titre_annonce.setText(annonce.getTitre());
        txt_text.setText(annonce.getTexte());
        txt_prix.setText(annonce.getPrix());
        txt_email.setText(annonce.getEmail());
        txt_tel.setText(annonce.getTel());
        txt_tel1.setText(annonce.getTel1());
        txt_tel2.setText(annonce.getTel2());
        txt_location.setText(annonce.getLocation());
        txt_categorie.setText(annonce.getCategorie());
        String str_share = this.getString(R.string.partager_annonce, annonce.getShare());
        btn_share.setText(str_share);

        String updatedAt = annonce.getUpdatedAt();
        String str_updatedAt = this.getString(R.string.derni_re_mise_jour_le_1_s, updatedAt);
        if (updatedAt != null) {
            if (!updatedAt.isEmpty()) {
                txt_updatedAt.setText(str_updatedAt);
                txt_updatedAt.setVisibility(View.VISIBLE);
            } else {
                txt_updatedAt.setVisibility(View.GONE);
            }
        }

        String datePub = getString(R.string.publi_le, annonce.getDate_pub());
        txt_date_pub.setText(datePub);

        if (annonce.getCategorie().equals("null") || annonce.getCategorie().isEmpty()) {
            txt_categorie.setText(R.string.aucune_categorie_renseignee);
        }

        if (annonce.getTel().isEmpty()) {
            txt_tel.setVisibility(View.GONE);
        }

        if (annonce.getTel1().isEmpty()) {
            txt_tel1.setVisibility(View.GONE);
        }

        if (annonce.getTel2().isEmpty()) {
            txt_tel2.setVisibility(View.GONE);
        }
        if (annonce.getEmail().length() == 0) {
            txt_email.setVisibility(View.GONE);
        }
        if (annonce.getPrix().length() == 0) {
            txt_prix.setText(R.string.prix_sur_demande);
        }
        if (annonce.getLocation().length() == 0) {
            txt_location.setVisibility(View.GONE);
        }
        if (annonce.getLocation().isEmpty()) {
            txt_location.setVisibility(View.GONE);
        }

        try {
            // JSONArray jsonArray = new JSONArray(annonce.getIllustrations());
            List<Announce.Annonce.Illustration> arrayList = annonce.illustrations;
            for (Announce.Annonce.Illustration data : arrayList) {
                imagesList.add(data.getNom());
                // Log.d(TAG, data.toString());
            }
            if (arrayList.size() == 0) {
                pager.setVisibility(View.GONE);
            }
            imagesAdapter.notifyDataSetChanged();

        } catch (Exception e) {
            e.printStackTrace();
        }

        btn_edit.setOnClickListener(v -> {
            try {
                Intent intent = new Intent(this, ActivityAnnonceEditer.class);
                intent.putExtra("annonce", annonce);
                intent.setFlags(Intent.FLAG_ACTIVITY_NO_HISTORY);
                startActivity(intent);
            } catch (Exception e) {
                e.printStackTrace();
            }
        });

        btn_edit_photo.setOnClickListener(v -> {
            try {
                Intent intent = new Intent(this, ActivityAnnounceEditImages.class);
                intent.putExtra("annonce", annonce);
                startActivity(intent);
            } catch (Exception e) {
                e.printStackTrace();
            }
        });

        btn_delete.setOnClickListener(v -> {
            AlertDialog.Builder builder = new AlertDialog.Builder(this);
            builder.setMessage(R.string.confirmation_suppression_annonce);
            builder.setPositiveButton(R.string.ok, (dialog, id) -> deleteAnnonce(annonce.getId_ann()));
            builder.setNegativeButton(R.string.annuler, (dialog, which) -> dialog.dismiss());
            AlertDialog dialog = builder.create();
            dialog.show();
        });

        btn_share.setOnClickListener(v -> {
            Intent sharingIntent = new Intent(Intent.ACTION_SEND);
            sharingIntent.setType("text/plain");
            String shareBodyText = annonce.getTexte() + "\n" + getString(R.string.telecharger_et_partager_l_application, "http://fasobizness.com/annonces/" + id_ann);
            sharingIntent.putExtra(Intent.EXTRA_SUBJECT, annonce.getTitre());
            sharingIntent.putExtra(Intent.EXTRA_TEXT, shareBodyText);
            startActivity(Intent.createChooser(sharingIntent, getString(R.string.partager_avec)));
        });


    }

    private void deleteAnnonce(String id_ann) {
        ProgressDialog progressDialog = new ProgressDialog(this);
        progressDialog.setTitle(getString(R.string.suppresion_en_cours));
        Call<MyResponse> call = api.deleteAnnounce(id_ann);
        call.enqueue(new Callback<MyResponse>() {
            @Override
            public void onResponse(@NonNull Call<MyResponse> call, @NonNull Response<MyResponse> response) {
                if (response.isSuccessful()) {
                    AlertDialog.Builder builder = new AlertDialog.Builder(ActivityDetailAnnonceUser.this);
                    builder.setMessage(R.string.annonce_supprimee_avec_succes);
                    builder.setPositiveButton(R.string.ok, (dialog, id) -> finish());
                    AlertDialog dialog = builder.create();
                    dialog.show();
                } else {
                    Toast.makeText(ActivityDetailAnnonceUser.this, R.string.une_erreur_sest_produite, Toast.LENGTH_SHORT).show();
                }
            }

            @Override
            public void onFailure(@NonNull Call<MyResponse> call, @NonNull Throwable t) {
                Toast.makeText(ActivityDetailAnnonceUser.this, R.string.pas_d_acces_internet, Toast.LENGTH_SHORT).show();
            }
        });
    }

    @Override
    public void onImageClicked(int position) {
        Intent intent = new Intent(this, ActivityFullScreen.class);
        intent.putStringArrayListExtra("images", imagesList);
        intent.putExtra("position", position);
        startActivity(intent);
    }

    @Override
    protected void onStart() {
        super.onStart();

        ann.setVisibility(View.GONE);
        loading_indicator_ann.setVisibility(View.VISIBLE);
        getAnnounce(id_ann);
    }
}

File: app/src/main/java/bf/fasobizness/bafatech/activities/annonce/ActivityDetailAnnonceUser.kt
package bf.fasobizness.bafatech.activities.annonce

//import android.app.ProgressDialog
import android.content.DialogInterface
import android.content.Intent
import android.os.Bundle
import android.util.Log
import android.view.View
import android.widget.Button
import android.widget.LinearLayout
import android.widget.TextView
import android.widget.Toast
import androidx.appcompat.app.AlertDialog
import androidx.appcompat.app.AppCompatActivity
import androidx.appcompat.widget.Toolbar
import androidx.viewpager.widget.ViewPager
import bf.fasobizness.bafatech.R
import bf.fasobizness.bafatech.activities.ActivityFullScreen
import bf.fasobizness.bafatech.adapters.ImagesAdapter
import bf.fasobizness.bafatech.helper.RetrofitClient
import bf.fasobizness.bafatech.interfaces.API
import bf.fasobizness.bafatech.interfaces.OnImageListener
import bf.fasobizness.bafatech.models.Announce.Annonce
import bf.fasobizness.bafatech.models.MyResponse
import bf.fasobizness.bafatech.utils.MySharedManager
import com.google.android.material.snackbar.Snackbar
import retrofit2.Call
import retrofit2.Callback
import retrofit2.Response
import java.util.*

class ActivityDetailAnnonceUser : AppCompatActivity(), OnImageListener {
    private lateinit var imagesList: ArrayList<String>
    private lateinit var imagesAdapter: ImagesAdapter
    private lateinit var user: String
    private lateinit var txtTitreAnnonce: TextView
    private lateinit var txtText: TextView
    private lateinit var txtPrix: TextView
    private lateinit var txtEmail: TextView
    private lateinit var txtTel1: TextView
    private lateinit var txtTel2: TextView
    private lateinit var txtTel: TextView
    private lateinit var txtLocation: TextView
    private lateinit var txtCategorie: TextView
    private lateinit var txtUpdatedat: TextView
    private lateinit var txtDatePub: TextView
    private lateinit var ann: LinearLayout
    private lateinit var layoutNoAnnonce: LinearLayout
    private lateinit var loadingIndicatorAnn: LinearLayout
    private lateinit var layoutEntOffline: LinearLayout
    private lateinit var layoutBusySystem: LinearLayout
    private lateinit var pager: ViewPager
    private lateinit var idAnn: String
    private lateinit var btnEdit: Button
    private lateinit var btnDelete: Button
    private lateinit var btnEditPhoto: Button
    private lateinit var btnShare: Button
    private var api: API = RetrofitClient.getClient().create(API::class.java)
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_detail_annonce_utilisateur)
        api
        val mySharedManager = MySharedManager(this)
        user = mySharedManager.user
        val toolbar = findViewById<Toolbar>(R.id.toolbar)
        toolbar.title = ""
        setSupportActionBar(toolbar)
        Objects.requireNonNull(supportActionBar)!!.setDisplayHomeAsUpEnabled(true)
        toolbar.setNavigationIcon(R.drawable.left_white)
        toolbar.setNavigationOnClickListener { finish() }
        txtTitreAnnonce = findViewById(R.id.txt_titre_annonce)
        txtText = findViewById(R.id.txt_texte_ann)
        txtPrix = findViewById(R.id.txt_prix_annonce)
        txtEmail = findViewById(R.id.txt_email_util)
        txtTel1 = findViewById(R.id.txt_tel1_annonce)
        txtTel2 = findViewById(R.id.txt_tel2_annonce)
        txtTel = findViewById(R.id.txt_tel_annonce)
        txtLocation = findViewById(R.id.txt_location_annonce)
        txtCategorie = findViewById(R.id.txt_categorie_annonce)
        btnShare = findViewById(R.id.btn_share)
        txtUpdatedat = findViewById(R.id.txt_date_modification)
        txtDatePub = findViewById(R.id.txt_date_pub)
        pager = findViewById(R.id.flipper_affiche_annonce)
        loadingIndicatorAnn = findViewById(R.id.loading_indicator_ann)
        layoutEntOffline = findViewById(R.id.layout_ent_offline)
        layoutNoAnnonce = findViewById(R.id.layout_no_annonce)
        ann = findViewById(R.id.ann)
        layoutBusySystem = findViewById(R.id.layout_busy_system)
        val btnRefresh = findViewById<Button>(R.id.btn_refresh)
        layoutEntOffline.visibility = View.GONE
        ann.visibility = View.GONE
        loadingIndicatorAnn.visibility = View.VISIBLE
        layoutNoAnnonce.visibility = View.GONE
        layoutBusySystem.visibility = View.GONE
        imagesList = ArrayList()
        imagesAdapter = ImagesAdapter(this, imagesList)
        pager.adapter = imagesAdapter
        imagesAdapter.setOnImageListener(this)
        val extras = intent
        if (extras.getStringExtra("id_ann") != null) {
            idAnn = extras.getStringExtra("id_ann")!!
            btnRefresh.setOnClickListener { getAnnounce(extras.getStringExtra("id_ann")) }
        }
        btnEdit = findViewById(R.id.btn_editer)
        btnDelete = findViewById(R.id.btn_delete)
        btnEditPhoto = findViewById(R.id.btn_editer_photo)
        btnShare = findViewById(R.id.btn_share)
    }

    private fun getAnnounce(id_ann: String?) {
        layoutEntOffline.visibility = View.GONE
        loadingIndicatorAnn.visibility = View.VISIBLE
        val call = api.getAnnounce(id_ann, user)
        call.enqueue(object : Callback<Annonce?> {
            override fun onResponse(call: Call<Annonce?>, response: Response<Annonce?>) {
                loadingIndicatorAnn.visibility = View.GONE
                if (response.isSuccessful) {
                    Log.d(TAG, response.toString())
                    loadingIndicatorAnn.visibility = View.GONE
                    ann.visibility = View.VISIBLE
                    val announce = response.body()
                    if (announce == null) {
                        layoutEntOffline.visibility = View.GONE
                        ann.visibility = View.GONE
                        layoutNoAnnonce.visibility = View.VISIBLE
                    } else {
                        populateData(announce)
                    }
                } else {
                    layoutBusySystem.visibility = View.VISIBLE
                }
            }

            override fun onFailure(call: Call<Annonce?>, t: Throwable) {
                layoutEntOffline.visibility = View.VISIBLE
                ann.visibility = View.GONE
                layoutNoAnnonce.visibility = View.GONE
                loadingIndicatorAnn.visibility = View.GONE
                Log.d(TAG, t.toString())
                Toast.makeText(this@ActivityDetailAnnonceUser, getString(R.string.pas_d_acces_internet), Toast.LENGTH_SHORT).show()
            }
        })
    }

    private fun populateData(annonce: Annonce) {
        txtTitreAnnonce.text = annonce.titre
        txtText.text = annonce.texte
        txtPrix.text = annonce.prix
        txtEmail.text = annonce.email
        txtTel.text = annonce.tel
        txtTel1.text = annonce.tel1
        txtTel2.text = annonce.tel2
        txtLocation.text = annonce.location
        txtCategorie.text = annonce.categorie
        val strShare = this.getString(R.string.partager_annonce, annonce.share)
        btnShare.text = strShare
        val updatedAt = annonce.updatedAt
        val strUpdatedat = this.getString(R.string.derni_re_mise_jour_le_1_s, updatedAt)
        if (updatedAt != null) {
            if (updatedAt.isNotEmpty()) {
                txtUpdatedat.text = strUpdatedat
                txtUpdatedat.visibility = View.VISIBLE
            } else {
                txtUpdatedat.visibility = View.GONE
            }
        }
        val datePub = getString(R.string.publiee_le, annonce.date_pub)
        txtDatePub.text = datePub
        if (annonce.categorie == "null" || annonce.categorie.isEmpty()) {
            txtCategorie.setText(R.string.aucune_categorie_renseignee)
        }
        if (annonce.tel.isEmpty()) {
            txtTel.visibility = View.GONE
        }
        if (annonce.tel1.isEmpty()) {
            txtTel1.visibility = View.GONE
        }
        if (annonce.tel2.isEmpty()) {
            txtTel2.visibility = View.GONE
        }
        if (annonce.email.isEmpty()) {
            txtEmail.visibility = View.GONE
        }
        if (annonce.prix.isEmpty()) {
            txtPrix.setText(R.string.prix_sur_demande)
        }
        if (annonce.location.isEmpty()) {
            txtLocation.visibility = View.GONE
        }
        if (annonce.location.isEmpty()) {
            txtLocation.visibility = View.GONE
        }
        try {
            // JSONArray jsonArray = new JSONArray(annonce.getIllustrations());
            val arrayList = annonce.illustrations
            for (data in arrayList) {
                imagesList.add(data.nom)
                // Log.d(TAG, data.toString());
            }
            if (arrayList.size == 0) {
                pager.visibility = View.GONE
            }
            imagesAdapter.notifyDataSetChanged()
        } catch (e: Exception) {
            e.printStackTrace()
        }
        btnEdit.setOnClickListener {
            try {
                val intent = Intent(this, ActivityAnnonceEditer::class.java)
                intent.putExtra("annonce", annonce)
                intent.flags = Intent.FLAG_ACTIVITY_NO_HISTORY
                startActivity(intent)
            } catch (e: Exception) {
                e.printStackTrace()
            }
        }
        btnEditPhoto.setOnClickListener {
            try {
                val intent = Intent(this, ActivityAnnounceEditImages::class.java)
                intent.putExtra("annonce", annonce)
                startActivity(intent)
            } catch (e: Exception) {
                e.printStackTrace()
            }
        }
        btnDelete.setOnClickListener {
            val builder = AlertDialog.Builder(this)
            builder.setMessage(R.string.confirmation_suppression_annonce)
            builder.setPositiveButton(R.string.ok) { _: DialogInterface?, _: Int -> deleteAnnonce(annonce.id_ann) }
            builder.setNegativeButton(R.string.annuler) { dialog: DialogInterface, _: Int -> dialog.dismiss() }
            val dialog = builder.create()
            dialog.show()
        }
        btnShare.setOnClickListener {
            val sharingIntent = Intent(Intent.ACTION_SEND)
            sharingIntent.type = "text/plain"
            // String shareBodyText = annonce.getTitre() + "\n" +annonce.getTexte() + "\n" + getString(R.string.telecharger_et_partager_l_application);
            val shareBodyText = """Salut, voici une annonce intéressante que je viens de découvrir sur Faso Biz Nèss : ${annonce.titre} ${annonce.texte}

Pour en savoir plus, clique ici : https://fasobizness.com/annonce/${annonce.id_ann}. 


Si tu n’as pas encore l’application tu peux la télécharger gratuitement sur Playstore : http://bit.ly/AndroidFBN"""
            sharingIntent.putExtra(Intent.EXTRA_SUBJECT, annonce.titre)
            sharingIntent.putExtra(Intent.EXTRA_TEXT, shareBodyText)
            startActivity(Intent.createChooser(sharingIntent, getString(R.string.partager_avec)))
            shareAnn(idAnn)
        }
    }

    private fun deleteAnnonce(id_ann: String) {
        /*val progressDialog = ProgressDialog(this)
        progressDialog.setTitle(getString(R.string.suppresion_en_cours))*/
        loadingIndicatorAnn.visibility = View.VISIBLE
        val call = api.deleteAnnounce(id_ann)
        call.enqueue(object : Callback<MyResponse?> {
            override fun onResponse(call: Call<MyResponse?>, response: Response<MyResponse?>) {
                if (response.isSuccessful) {
                    val builder = AlertDialog.Builder(this@ActivityDetailAnnonceUser)
                    builder.setMessage(R.string.annonce_supprimee_avec_succes)
                    builder.setPositiveButton(R.string.ok) { _: DialogInterface?, _: Int -> finish() }
                    val dialog = builder.create()
                    dialog.show()
                } else {
                    Toast.makeText(this@ActivityDetailAnnonceUser, R.string.une_erreur_sest_produite, Toast.LENGTH_SHORT).show()
                }
            }

            override fun onFailure(call: Call<MyResponse?>, t: Throwable) {
                Toast.makeText(this@ActivityDetailAnnonceUser, R.string.pas_d_acces_internet, Toast.LENGTH_SHORT).show()
            }
        })
    }

    override fun onImageClicked(position: Int) {
        val intent = Intent(this, ActivityFullScreen::class.java)
        intent.putStringArrayListExtra("images", imagesList)
        intent.putExtra("position", position)
        startActivity(intent)
    }

    private fun shareAnn(id_ann: String) {
        val call = api.setAnnouncesActions("share", id_ann, user)
        call.enqueue(object : Callback<MyResponse?> {
            override fun onResponse(call: Call<MyResponse?>, response: Response<MyResponse?>) {
                Log.d("AnnonceDetails", response.body().toString())
            }

            override fun onFailure(call: Call<MyResponse?>, t: Throwable) {
                Toast.makeText(this@ActivityDetailAnnonceUser, R.string.pas_d_acces_internet, Toast.LENGTH_SHORT).show()
            }
        })
    }

    override fun onStart() {
        super.onStart()
        ann.visibility = View.GONE
        loadingIndicatorAnn.visibility = View.VISIBLE
        getAnnounce(idAnn)
    }

    companion object {
        private const val TAG = "ActivityDetailsAnnonce"
    }
}